### ISSUE TITLE
Filter to mimic DS screen colors

### ISSUE THREAD

**Issue by Brankale** [Arisotura/melonDS#1179]
It's an odd request but it would be awesome to have a filter to mimic the DS Lite screen.

I ask for this because DS Lite lcd screen (and maybe the original DS and DSi) seems to have more vivid colors.
It's not a big problem but in some games there's a really big difference.

This is a comparison between a DS Lite lcd screen (the first) and 2DS lcd screen (the second):

![ds_lite](https://user-images.githubusercontent.com/45881310/129273810-6bf22c62-8321-462c-8c6e-2818a711db73.jpg)
![2ds](https://user-images.githubusercontent.com/45881310/129273817-7fa4f447-36c3-4330-a128-19131b598567.jpg)

They are not great photos but at least they can give you an idea.

The 2DS screen has colors which are definitly washed out compared to the DS Lite.
I also calibrated my monitor in the best way I could but is much closer to the 2DS screen.

I don't know if it's just an effect due to the low resolution and the black gap between pixels but in the end colors looks different.

**Comment by Arisotura** [comment]
well, melonDS renders the colors are they are provided by the game. it would be possible to apply color effects, but I'm not sure how to make it look more like a DSLite screen -- imo DSLite screens already have pretty good colors. I could look more in depth into this, this isn't a high priority atm tho.

**Comment by MojoJojoDojo** [comment]
No$gba got something like that I think.
You can put the screen settings to be similar to the DS Lite and it will also change the colors as well.

**Comment by Brankale** [comment]
> No$gba got something like that I think.

Yes, you are right, I've never noticed it. On No$gba is more similar to the DS Lite lcd screen.

I took some screenshots just for comparison:

No$gba (DS Lite):
![DS Lite](https://user-images.githubusercontent.com/45881310/129328958-94e5676a-51b6-4054-aa41-160e04d9e747.png)

MelonDS
![melonDS](https://user-images.githubusercontent.com/45881310/129328932-9333d954-f43d-4527-a74d-d3e2e75d2538.png)

Color temperature is different

**Comment by Brankale** [comment]
I'm playing with colour spaces conversions and I'm getting somewhere near to the correct result but I need the correct values for the gamma of every channel (i.e. red, green, blue) in order to do a good job and also not to go crazy. Currently I have only the gamma for the greyscale but it's not enough or so I think. So if anyone knows these values, please write them here.

**Comment by Brankale** [comment]
I wrote a converter which, in theory, should be able to modify image colors to reproduce DS Lite screen colors.
It's not perfect for a varierity of reasons:
- i don't know the exact gamma for every channel
- my best monitor has an average DeltaE of 1.24 which is good but not perfect
- etc

I uploaded a Java program (both JAR and source code if you don't trust the JAR):
[converter.zip](https://github.com/Arisotura/melonDS/files/7656193/converter.zip)

To convert an image you should do:
`java -jar JAR_PATH IMAGE_PATH` from the command line

I would appreciate if you can try it and tell me if it's similar to a DS Lite screen.
NB: Your screen must be well calibrated according to the sRGB standard otherwise colours will look totally different!

NB: If I remember correctly, I have seen from a friend of mine who has a DSi that colors are similar to DS Lite, so if you have a DSi you can use this program.

**Comment by Brankale** [comment]
I'll post some images here to show the output of the converter:

MelonDS output on the left / Converted image on the right:

![green_path](https://user-images.githubusercontent.com/45881310/144853905-cd1d1891-9920-47b7-9bd8-e162af33db00.PNG) ![green_path](https://user-images.githubusercontent.com/45881310/144853970-792e9678-fe9c-4ca1-bdb2-d657ba66d14a.PNG)

![team](https://user-images.githubusercontent.com/45881310/144854268-4ec94fd5-84fa-4ade-ad07-30e802faf777.PNG) ![team](https://user-images.githubusercontent.com/45881310/144854275-40ac8de9-3fa5-4e4d-8527-bd33caf9b86b.PNG)

![red_floor](https://user-images.githubusercontent.com/45881310/144854330-2f5089d6-d9d6-44f5-9da7-a6ceccd9fa46.PNG) ![red_floor](https://user-images.githubusercontent.com/45881310/144854335-54ebe0c7-ff93-4e40-83a0-2d6cf50cec8a.PNG)

**Comment by Brankale** [comment]
**Update**: I started a project to accurately reproduce the console's screen colors using actual measurement tools, such as colorimeters, to get accurate screen data. You can find it here: https://github.com/Brankale/Handheld-Color-Space-Project

it's still work in progress in some areas and the documentation must be revised a little bit but overall it's in good shape.

I'll leave you a preview here

| Raw output | Shader |
| :-------------: | :-------------: | 
| ![DS no shader](https://github.com/user-attachments/assets/8e7f0c11-3dab-4217-95f5-f24c085129bc) <br>`raw output`  | ![DS Phat](https://github.com/user-attachments/assets/503444a4-13e0-4f75-a839-59cd8150d3b6) <br>`DS Phat` |
| ![DS no shader](https://github.com/user-attachments/assets/8e7f0c11-3dab-4217-95f5-f24c085129bc) <br>`raw output`  | ![DS Lite](https://github.com/user-attachments/assets/98afdda1-5ec9-4b34-9347-cd5a59aa9cd7) <br>`DS Lite` |
| ![DS no shader](https://github.com/user-attachments/assets/8e7f0c11-3dab-4217-95f5-f24c085129bc) <br>`raw output` | ![3DS](https://github.com/user-attachments/assets/97e85339-214b-4dac-8d3c-849498e91a60) <br>`3DS` |
| ![DS no shader](https://github.com/user-attachments/assets/8e7f0c11-3dab-4217-95f5-f24c085129bc) <br>`raw output` | ![New 3DS XL](https://github.com/user-attachments/assets/906c5e60-93b1-4a06-bb2a-8a89a5bf30b6) <br>`New 3DS XL` |
